test_tools/src/test_tools/odometer.py

```diff
@@ -140,25 +146,30 @@ def write(self):
         Writing all data in self.data to the data file and closing it again. This should prevent file corruption.
         :return: no return
         """
-        # Create today's file if not already there
-        if os.path.exists(self.newfilepath):
-            new_file = open(self.newfilepath, "a", 1)  # 1=line-buffered
-            rospy.logdebug("Today's file already exists")
-        else:
-            new_file = open(self.newfilepath, "w+", 1)  # 1=line-buffered
-            rospy.logdebug("First time writing in today's file")
+        try:
+            # Create today's file if not already there
+            if os.path.exists(self.newfilepath):
+                new_file = open(self.newfilepath, "a", 1)  # 1=line-buffered
+                rospy.logdebug("Today's file already exists")
+            else:
+                new_file = open(self.newfilepath, "w+", 1)  # 1=line-buffered
+                rospy.logdebug("First time writing in today's file")
```

Comment:
```python
try:
    if os.path.exists(self.newfilepath):
        rospy.logdebug("Today's file already exists")
    else:
        rospy.logdebug("First time writing in today's file")

    with open(self.newfilepath, "a", 1) as newfile:
```

This will automatically close the file even if exception occurs

Reply:
Should the `try` not be within the `with`?